File: tugas_1.py
KAPASITAS = 10
nim = []
nama = []

def menu():
    print("\n=== DATA MAHASISWA ===")
    print("1. Insert at beginning")
    print("2. Insert at given position")
    print("3. Insert at end")
    print("4. Delete from beginning")
    print("5. Delete given position")
    print("6. Delete from end")
    print("7. Delete first occurrence")
    print("8. Show data")
    print("9. Exit")

def insert_beginning():
    if len(nim) == KAPASITAS:
        print("Array penuh!")
        return

    n = input("Masukkan NIM  : ")
    nm = input("Masukkan Nama : ")

    nim.insert(0, n)
    nama.insert(0, nm)

def insert_at_position():
    if len(nim) == KAPASITAS:
        print("Array penuh!")
        return

    try:
        pos = int(input(f"Masukkan index (0 - {len(nim)}): "))
    except:
        print("Input harus angka!")
        return

    if pos < 0 or pos > len(nim):
        print("Index tidak valid!")
        return

    n = input("Masukkan NIM  : ")
    nm = input("Masukkan Nama : ")

    nim.insert(pos, n)
    nama.insert(pos, nm)

def insert_end():
    if len(nim) == KAPASITAS:
        print("Array penuh!")
        return

    n = input("Masukkan NIM  : ")
    nm = input("Masukkan Nama : ")

    nim.append(n)
    nama.append(nm)

def delete_beginning():
    if len(nim) == 0:
        print("Data kosong!")
        return

    nim.pop(0)
    nama.pop(0)

def delete_at_position():
    if len(nim) == 0:
        print("Data kosong!")
        return

    try:
        pos = int(input(f"Masukkan index (0 - {len(nim)-1}): "))
    except:
        print("Input harus angka!")
        return

    if pos < 0 or pos >= len(nim):
        print("Index tidak valid!")
        return

    nim.pop(pos)
    nama.pop(pos)

def delete_end():
    if len(nim) == 0:
        print("Data kosong!")
        return

    nim.pop()
    nama.pop()

def delete_first_occurrence():
    if len(nim) == 0:
        print("Data kosong!")
        return

    target = input("Masukkan NIM yang ingin dihapus: ")

    if target in nim:
        index = nim.index(target)
        nim.pop(index)
        nama.pop(index)
    else:
        print("Data tidak ditemukan!")

def show_data():
    if len(nim) == 0:
        print("Data kosong!")
        return

    print("\nIndex | NIM         | Nama")
    for i in range(len(nim)):
        print(f"{i}     | {nim[i]} | {nama[i]}")

# Program utama
while True:
    menu()
    try:
        pilihan = int(input("Pilih Opsi: "))
    except:
        print("Input harus angka!")
        continue

    if pilihan == 1:
        insert_beginning()
    elif pilihan == 2:
        insert_at_position()
    elif pilihan == 3:
        insert_end()
    elif pilihan == 4:
        delete_beginning()
    elif pilihan == 5:
        delete_at_position()
    elif pilihan == 6:
        delete_end()
    elif pilihan == 7:
        delete_first_occurrence()
    elif pilihan == 8:
        show_data()
    elif pilihan == 9:
        print("Keluar dari program...")
        break
    else:
        print("Pilihan tidak valid!")

File: tugas_1.java
import java.util.Scanner;

public class tugas_1 {

    static final int KAPASITAS = 10;
    static String[] nim = new String[KAPASITAS];
    static String[] nama = new String[KAPASITAS];
    static int jumlahData = 0;

    static Scanner input = new Scanner(System.in);

    public static void main(String[] args) {
        int pilihan;

        do {
            menu();
            pilihan = input.nextInt();
            input.nextLine(); // buang newline

            switch (pilihan) {
                case 1:
                    insertBeginning();
                    break;
                case 2:
                    insertAtPosition();
                    break;
                case 3:
                    insertEnd();
                    break;
                case 4:
                    deleteBeginning();
                    break;
                case 5:
                    deleteAtPosition();
                    break;
                case 6:
                    deleteEnd();
                    break;
                case 7:
                    deleteFirstOccurrence();
                    break;
                case 8:
                    showData();
                    break;
                case 9:
                    System.out.println("Keluar dari program...");
                    break;
                default:
                    System.out.println("Pilihan tidak valid!");
            }
        } while (pilihan != 9);
    }

    static void menu() {
        System.out.println("\n=== DATA MAHASISWA ===");
        System.out.println("1. Insert at beginning (index 0)");
        System.out.println("2. Insert at given position (index array)");
        System.out.println("3. Insert at end");
        System.out.println("4. Delete from beginning (index 0)");
        System.out.println("5. Delete given position (index array)");
        System.out.println("6. Delete from end");
        System.out.println("7. Delete first occurrence (berdasarkan NIM)");
        System.out.println("8. Show data");
        System.out.println("9. Exit");
        System.out.print("Pilih menu: ");
    }

    static void insertBeginning() {
        if (jumlahData == KAPASITAS) {
            System.out.println("Array penuh!");
            return;
        }

        System.out.print("Masukkan NIM  : ");
        String n = input.nextLine();
        System.out.print("Masukkan Nama : ");
        String nm = input.nextLine();

        for (int i = jumlahData; i > 0; i--) {
            nim[i] = nim[i - 1];
            nama[i] = nama[i - 1];
        }

        nim[0] = n;
        nama[0] = nm;
        jumlahData++;
    }

    static void insertAtPosition() {
        if (jumlahData == KAPASITAS) {
            System.out.println("Array penuh!");
            return;
        }

        System.out.print("Masukkan index (0 - " + jumlahData + "): ");
        int pos = input.nextInt();
        input.nextLine();

        if (pos < 0 || pos > jumlahData) {
            System.out.println("Index tidak valid!");
            return;
        }

        System.out.print("Masukkan NIM  : ");
        String n = input.nextLine();
        System.out.print("Masukkan Nama : ");
        String nm = input.nextLine();

        for (int i = jumlahData; i > pos; i--) {
            nim[i] = nim[i - 1];
            nama[i] = nama[i - 1];
        }

        nim[pos] = n;
        nama[pos] = nm;
        jumlahData++;
    }

    static void insertEnd() {
        if (jumlahData == KAPASITAS) {
            System.out.println("Array penuh!");
            return;
        }

        System.out.print("Masukkan NIM  : ");
        nim[jumlahData] = input.nextLine();
        System.out.print("Masukkan Nama : ");
        nama[jumlahData] = input.nextLine();

        jumlahData++;
    }

    static void deleteBeginning() {
        if (jumlahData == 0) {
            System.out.println("Data kosong!");
            return;
        }

        for (int i = 0; i < jumlahData - 1; i++) {
            nim[i] = nim[i + 1];
            nama[i] = nama[i + 1];
        }

        jumlahData--;
    }

    static void deleteAtPosition() {
        if (jumlahData == 0) {
            System.out.println("Data kosong!");
            return;
        }

        System.out.print("Masukkan index (0 - " + (jumlahData - 1) + "): ");
        int pos = input.nextInt();
        input.nextLine();

        if (pos < 0 || pos >= jumlahData) {
            System.out.println("Index tidak valid!");
            return;
        }

        for (int i = pos; i < jumlahData - 1; i++) {
            nim[i] = nim[i + 1];
            nama[i] = nama[i + 1];
        }

        jumlahData--;
    }

    static void deleteEnd() {
        if (jumlahData == 0) {
            System.out.println("Data kosong!");
            return;
        }

        jumlahData--;
    }

    static void deleteFirstOccurrence() {
        if (jumlahData == 0) {
            System.out.println("Data kosong!");
            return;
        }

        System.out.print("Masukkan NIM yang ingin dihapus: ");
        String target = input.nextLine();

        int index = -1;
        for (int i = 0; i < jumlahData; i++) {
            if (nim[i].equals(target)) {
                index = i;
                break;
            }
        }

        if (index == -1) {
            System.out.println("Data tidak ditemukan!");
            return;
        }

        for (int i = index; i < jumlahData - 1; i++) {
            nim[i] = nim[i + 1];
            nama[i] = nama[i + 1];
        }

        jumlahData--;
    }

    static void showData() {
        if (jumlahData == 0) {
            System.out.println("Data kosong!");
            return;
        }

        System.out.println("\nIndex | NIM        | Nama");
        for (int i = 0; i < jumlahData; i++) {
            System.out.println(i + "     | " + nim[i] + " | " + nama[i]);
        }
    }
}
